internal/olm/operator/internal/registry_pod.go

```diff
@@ -0,0 +1,145 @@
+package olm
+
+import (
+	"bytes"
+	"context"
+	"flag"
+	"fmt"
+	"path/filepath"
+	"time"
+
+	core "k8s.io/api/core/v1"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/util/homedir"
+)
+
+// RegistryPod holds resources necessary for creation of a registry server
+type RegistryPod struct {
+	registryImage       string
+	registryPodGRPCPort int32
+	registryDBPath      string
+	registryBundleImage string
+
+	ctx        context.Context
+	kubeclient kubernetes.Interface
+}
+
+// GetKubeClient creates a kubernetes clientset using kubeconfig
+func GetKubeClient() (client kubernetes.Interface) {
+	var kubeconfig *string
+	if home := homedir.HomeDir(); home != "" { // Check if machine has home directory
+		// Read kubeconfig flag. If not provided, use config file $HOME/.kube/config
+		kubeconfig = flag.String("kubeconfig", filepath.Join(home, ".kube", "config"), "(optional) absolute path to the kubeconfig file")
+	} else {
+		kubeconfig = flag.String("kubeconfig", "", "absolute path to the kubeconfig file")
+	}
+	flag.Parse()
+
+	// Build configuration from the config file
+	config, err := clientcmd.BuildConfigFromFlags("", *kubeconfig)
+	if err != nil {
+		panic(err)
+	}
+
+	// Create kubernetes clientset, which can be used to create, delete, patch, list kubernetes resources
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		panic(err)
+	}
+
+	return clientset
+}
+
+// CreateBundleRegistry returns a registry pod built from an index image
+func (rp *RegistryPod) CreateBundleRegistry() *core.Pod {
+	var pod *core.Pod
```

Comment:
Before building the pod, we should set defaults for empty fields. E.g.

```go
const (
  defaultIndexImage = "quay.io/operator-framework/upstream-opm-builder:latest"
)

...

func (rp *RegistryPod) setDefaults() {
  if len(rp.IndexImage) == 0 {
    rp.IndexImage = defaultIndexImage
  }
  if len(rp.BundleAddMode) == 0 {
    if rp.IndexImage == defaultIndexImage {
      rp.BundleAddMode = "semver"
    } else {
      rp.BundleAddMode = "replaces"
    }
  }
}
```

And then validate the struct, e.g.:
```go
func (rp *RegistryPod) validate() error {
  if len(rp.BundleImage) == 0 {
    return fmt.Errorf("bundle image is required")
  }
  if len(rp.DBPath) == 0 {
    return fmt.Errorf("registry database path is required")
  }
  return nil
}
```